keel-sql/src/main/kotlin/com/netflix/spinnaker/keel/sql/SqlUnhappyVetoRepository.kt

```diff
@@ -51,18 +68,18 @@ class SqlUnhappyVetoRepository(
     }
   }
 
-  override fun getOrCreateVetoStatus(resourceId: String, application: String, wait: Duration): UnhappyVetoStatus {
+  override fun getOrCreateVetoStatus(resourceId: String, application: String, wait: Duration?): UnhappyVetoStatus {
     sqlRetry.withRetry(READ) {
-      jooq
-        .select(UNHAPPY_VETO.RECHECK_TIME)
-        .from(UNHAPPY_VETO)
-        .where(UNHAPPY_VETO.RESOURCE_ID.eq(resourceId))
-        .fetchOne()
-    }
-      ?.let { (recheckTime) ->
+        jooq
+          .select(UNHAPPY_VETO.RECHECK_TIME)
+          .from(UNHAPPY_VETO)
+          .where(UNHAPPY_VETO.RESOURCE_ID.eq(resourceId))
+          .fetchOne()
+      }
+      ?.let { (recheckTime: LocalDateTime?) ->
         return UnhappyVetoStatus(
-          shouldSkip = recheckTime > clock.timestamp(),
-          shouldRecheck = recheckTime < clock.timestamp()
+          shouldSkip = recheckTime == null || recheckTime > clock.timestamp(),
+          shouldRecheck = recheckTime != null && recheckTime < clock.timestamp()
```

Comment:
`recheckTime <= clock.timestamp()`? (One of them has to be inclusive, right?)

Reply:
You'd think, but I didn't change the logic.

Reply:
Oh boy...

Reply:
Look at the code, it's not obvious to me why `shouldRecheck` wasn't defined as `!shouldSkip` (or vice versa) in the first place. I understand the desire to not change existing behavior, but in a future PR I think maybe we should revisit this.

@emjburns Do you have any insight here?

Reply:
I've updated this now to match the query. As Luis pointed out, it was inconsistent.